### PR TITLE
Arabic

### PR DESCRIPTION
**Pull request by anand-nv**
# What does this PR do ?
Add Arabic TN and ITN along with tests to Nemo-text-processing

  
**PR Type**:
- [ ] New Feature: Adds support for Arabic TN/ITN

### CODE REVIEW

File: nemo_text_processing/inverse_text_normalization/ar/taggers/decimal.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
```
Comment (yzhang123) [inline nemo_text_processing/inverse_text_normalization/ar/taggers/decimal.py:1]: could you update the top line of all headers to 
Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.

File: nemo_text_processing/inverse_text_normalization/ar/taggers/punctuation.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
```
Comment (yzhang123) [inline nemo_text_processing/inverse_text_normalization/ar/taggers/punctuation.py:2]: remove this line please

File: nemo_text_processing/inverse_text_normalization/ar/taggers/word.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
```
Comment (yzhang123) [inline nemo_text_processing/inverse_text_normalization/ar/taggers/word.py:2]: remove this line

File: nemo_text_processing/inverse_text_normalization/ar/verbalizers/money.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    def __init__(self, decimal: GraphFst, deterministic: bool = True):
+        super().__init__(name="money", kind="verbalize", deterministic=deterministic)
+        unit = (
+            pynutil.delete("currency:")
+            + delete_space
+            + pynutil.delete("\"")
+            + pynini.closure(NEMO_CHAR - " ", 1)
+            + pynutil.delete("\"")
+        )
+        # optionl_sign = pynini.closure(pynini.cross("negative: \"true\"", "-") + delete_space, 0, 1)
```
Comment (yzhang123) [inline nemo_text_processing/inverse_text_normalization/ar/verbalizers/money.py:38]: remove these comments?

File: nemo_text_processing/inverse_text_normalization/ar/verbalizers/verbalize_final.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
```
Comment (yzhang123) [inline nemo_text_processing/inverse_text_normalization/ar/verbalizers/verbalize_final.py:2]: remove this line. we only need this for graphfst file

File: nemo_text_processing/inverse_text_normalization/ar/taggers/cardinal.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
```
Comment (yzhang123) [inline nemo_text_processing/inverse_text_normalization/ar/taggers/cardinal.py:2]: remove this line

File: nemo_text_processing/inverse_text_normalization/ar/verbalizers/word.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+# Copyright 2015 and onwards Google, Inc.
```
Comment (yzhang123) [inline nemo_text_processing/inverse_text_normalization/ar/verbalizers/word.py:2]: remove line

File: tests/nemo_text_processing/ar/data_text_normalization/test_cases_money.txt [hunk 8]
```diff
@@ -0,0 +1,26 @@
+$7~سبعة دولار
+$7.50~سبعة دولار وخمسون سنت
+$7.05~سبعة دولار وخمسة سنتات
+$7.99~سبعة دولار وتسعة وتسعون سنت
+$7.10~سبعة دولار وعشرة سنتات
+$0.99~تسعة وتسعون سنت
```
Comment (yzhang123) [inline tests/nemo_text_processing/ar/data_text_normalization/test_cases_money.txt:6]: what about other classes like date, time, whitelist etc?

**Comment by yzhang123** [comment]
@anand-nv this is great, thank you.
 most of suggestions I added apply to multiple files, but i might have just mentioned it in one comment.

File: tests/nemo_text_processing/ar/data_text_normalization/test_cases_money.txt (hunk 8 above)
Comment (anand-nv) [reply tests/nemo_text_processing/ar/data_text_normalization/test_cases_money.txt:6]: Date/Time are not supported. Whitelist, will add them.

File: nemo_text_processing/text_normalization/ar/graph_utils.py [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+
+import os
+import string
+from pathlib import Path
+from typing import Dict
+
+from nemo_text_processing.text_normalization.en.utils import get_abs_path
+
+try:
+    import pynini
```
[the remark this answers is not in the document]
Comment (anand-nv) [reply nemo_text_processing/text_normalization/ar/graph_utils.py:23]: Done

File: nemo_text_processing/inverse_text_normalization/ar/verbalizers/word.py (hunk 7 above)
Comment (anand-nv) [reply nemo_text_processing/inverse_text_normalization/ar/verbalizers/word.py]: Done

File: nemo_text_processing/inverse_text_normalization/ar/taggers/cardinal.py (hunk 6 above)
Comment (anand-nv) [reply nemo_text_processing/inverse_text_normalization/ar/taggers/cardinal.py]: Done

File: nemo_text_processing/text_normalization/ar/data/money/local_currency.tsv [hunk 10]
```diff
@@ -0,0 +1,20 @@
+ج.م	جنيه مصري
+د.ع	دينار عراقي
+د.ك	دينار كويتي
+أ.م	أوقية موريتانية
```
[the remark this answers is not in the document]
Comment (anand-nv) [reply nemo_text_processing/text_normalization/ar/data/money/local_currency.tsv:4]: __init__.py exists

File: nemo_text_processing/text_normalization/ar/graph_utils.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (yzhang123) [reply nemo_text_processing/text_normalization/ar/graph_utils.py:23]: there is still exception for this file

**Review by yzhang123 — CHANGES_REQUESTED** [review]
please 
1. change header to 2023 and add AFFLIATES
2. keep google in header only for file graph_utils.py 
3. pls add whitelist test and files
4. some classes are missing (date, time, ordinal, ..)
5. add arabic support to sparrowhawk in tools/...
6. follow code analyzer and remove unused  variables and imports (when you go to files changed in github it will show you the warning at respective code lin)

**Review by yzhang123 — APPROVED** [review]

**Pull request merged**